ROADMAP.md

```diff
@@ -74,12 +74,13 @@ Using `gfx-rs` for backend abstraction allows us to provide compatibility with m
 **Next steps**
  * Raw input support (winit?)
  * Gamepad support (winit?)
- * Input Mapping
+ * Input Mapping: Map input events (e.g key press) to abstract action events (`Jump`, `Forward`, ..)
```

Comment:
This sort of can be done now. You can catch keyboard events in `State::handle_events` method and then republish them as abstracts events using `Broadcaster`. Like this:
```rust
fn handle_events(&mut self, events: &[WindowEvent], _: &mut World, _: &mut AssetManager, _: &mut Pipeline) -> Trans {
    use amethyst::event::*;
    let broadcaster = world.write_resource::<Broadcaster>();
    for event in events {
        let movement_event = match event.payload {
            Event::KeyboardInput(_, _, Some(VirtualKeyCode::W)) => Some(MovementEvent::Up),
            Event::KeyboardInput(_, _, Some(VirtualKeyCode::A)) => Some(MovementEvent::Left),
            Event::KeyboardInput(_, _, Some(VirtualKeyCode::S)) => Some(MovementEvent::Down),
            Event::KeyboardInput(_, _, Some(VirtualKeyCode::D)) => Some(MovementEvent::Right),
            _ => None,
        }
        if let Some(movement_event) = movement_event {
            broadcaster.publish(movement_event);
        }
        match event.payload {
            Event::KeyboardInput(_, _, Some(VirtualKeyCode::Escape)) => return Trans::Quit,
            Event::Closed => return Trans::Quit,
            _ => (),
        }
    }
    Trans::None
}
```
`MovementEvent` enum must have a `Component` trait impl and must be registered with `Broadcaster`, also `Broadcaster` must be added to the `world`. Registering `MovementEvent`/adding `Broadcaster` can be done in `State::on_start`.